pkg/asset/agent/image/agentimage.go

```diff
@@ -160,10 +192,28 @@ func (a *AgentImage) PersistToFile(directory string) error {
 	// Remove symlink if it exists
 	os.Remove(agentIsoFile)
 
-	err := isoeditor.Create(agentIsoFile, a.tmpPath, a.volumeID)
-	if err != nil {
-		return err
+	var err error
+	// For external platform when the bootArtifactsBaseUrl is specified,
+	// output the rootfs file alongside the minimal ISO
+	if a.platform == external.Name && a.bootArtifactsBaseURL != "" {
+		bootArtifactsFullPath := filepath.Join(directory, bootArtifactsPath)
+		err := extractRootFS(bootArtifactsFullPath, a.tmpPath, a.cpuArch)
+		if err != nil {
+			return err
+		}
+		logrus.Infof("RootFS file created in: %s. Upload it at %s", bootArtifactsFullPath, a.rootFSURL)
+		err = isoeditor.CreateMinimalISO(a.tmpPath, a.volumeID, a.rootFSURL, a.cpuArch, agentIsoFile)
```

Comment:
This part should not be conditional on bootArtifactsBaseURL being set. If it is not set we still want to generate a minimal ISO, just using the default rootfs URL. So the if statement on line 198 needs to be split in two, and the  `if a.bootArtifactsBaseURL != ""` one should only wrap the extractRootFS() part and not the CreateMinimalISO() part.